packages/rule-meta-theme-color/src/rule.ts

```diff
@@ -0,0 +1,184 @@
+             *
+             *  From: https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color
+             *
+             *  " The element has a name attribute, whose value is
+             *    an ASCII case-insensitive match for `theme-color` "
+             */
+
+            const nameAttributeValue = element.getAttribute('name');
+
+            if (nameAttributeValue && nameAttributeValue !== nameAttributeValue.trim()) {
+                await context.report(resource, element, `'name' attribute needs to be 'theme-color' (not '${nameAttributeValue}')`);
+            }
+        };
+
+        const checkContentAttributeValue = async (resource: string, element: IAsyncHTMLElement) => {
+            const contentValue = element.getAttribute('content');
+            const normalizedContentValue = normalizeString(contentValue, '');
+            const color = parseColor(normalizedContentValue);
+
+            if (color === null) {
+                await context.report(resource, element, `'content' attribute value ('${contentValue}') is invalid`);
+
+                return;
+            }
+
+            /*
+             * `theme-color` can accept any CSS `<color>`:
+             *
+             *    * https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color
+             *    * https://drafts.csswg.org/css-color/#typedef-color
+             *
+             *  However, `HWB` and `RGBA` value are not supported
+             *  everywhere `theme-color` is. Also, values such as
+             *  `currentcolor` don't make sense, but they will be
+             *  catched by the above check.
+             *
+             *  See also:
+             *
+             *    * https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#Browser_compatibility
+             *    * https://cs.chromium.org/chromium/src/third_party/WebKit/Source/platform/graphics/Color.cpp?rcl=6263bcf0ec9f112b5f0d84fc059c759302bd8c67
+             */
+
+            const targetedBrowsers: string = context.targetedBrowsers.join();
+            const rgbaIsSupported = targetedBrowsers && isSupported('css-rrggbbaa', targetedBrowsers);
+            const rgbaRegex = /^#([0-9a-fA-F]{4}){1,2}$/;
+
+            if (
+                // `HWB` is not supported anywhere (?).
+                color.model === 'hwb' ||
+
+                // `RGBA` support depends on the browser.
+                (color.model === 'rgb' &&
+                    rgbaRegex.test(normalizedContentValue) &&
+                    !rgbaIsSupported)
+            ) {
+                await context.report(resource, element, `'content' attribute value ('${contentValue}') is not unsupported`);
+            }
+        };
+
+        const validate = async (event: TraverseEnd) => {
+            const { resource }: { resource: string } = event;
+
+            // The following check don't make sense for non-HTML documents.
+
+            if (!isHTMLDocument(resource, context.pageHeaders)) {
```

Comment:
Isn't `TraverseEnd` send only for HTML documents? If so this check is not needed.

Reply:
Any reason why you are not subscribed to `traverse::element::meta`? This way you are sure to only receive the element for HTML, you can remove parts of the code and you only need to keep track if a previous `theme-color` has been emitted already.

You only need to subscribe to `traverse::end` and if none found previously report it.

Reply:
Done.